lib/handler/proxy.c

```diff
@@ -96,6 +96,7 @@ static void on_context_init(h2o_handler_t *_self, h2o_context_t *ctx)
         client_ctx->websocket_timeout = NULL;
     }
     client_ctx->ssl_ctx = self->config.ssl_ctx;
+    client_ctx->ssl_session_cache = h2o_cache_create(0, 4096, 86400 * 1000, h2o_socket_ssl_destroy_session_cache_entry);
```

Comment:
Oh, I missed the fact that the function is thread-specific.

So let me rephrase the question to: why are we creating a cache per every thread? I believe it should be shared between the worker threads.